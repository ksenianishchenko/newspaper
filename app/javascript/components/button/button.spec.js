import React from "react";
import { shallow } from "enzyme";
import Button from "./button";

describe("Button", () => {

  function createButton(type, onClick, className) {
    const btn = shallow(
      <Button type={type}
      className={className}
      onClick={onClick}>Click me</Button>
    )
    return btn;
  };

  const mockFn = jest.fn();

  it("should be render correctly", () => {

    const tree = createButton("submit", mockFn, "button");

    expect(tree).toMatchSnapshot();
  })

  it("should call mock function on click", () => {

    const tree = createButton("submit", mockFn, "button");

    tree.simulate("click");
    expect(mockFn).toHaveBeenCalled();
  })

  it("should render default class", () => {
    const tree = createButton("submit", mockFn);

    expect(tree.hasClass('btn btn-primary btn-block waves-effect waves-light')).toBe(true);
  })

});
