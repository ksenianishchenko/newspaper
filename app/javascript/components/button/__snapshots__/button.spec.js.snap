// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should be render correctly 1`] = `
<button
  className="button"
  onClick={[MockFunction]}
  type="submit"
>
  Click me
</button>
`;
